src/matcha_ml/services/global_parameters_service.py

```diff
@@ -0,0 +1,124 @@
+"""Global parameter service for creating and modifying a users global config files."""
+import os
+import uuid
+from typing import Any, Dict, Optional
+
+import yaml
+
+from matcha_ml.errors import MatchaPermissionError
+
+
+class GlobalParameters:
+    """A Global parameters service for interacting and updating a users global config file.
+
+    Users are opted-in for analytics data collection by default.
+    """
+
+    _instance: Optional["GlobalParameters"] = None
+    _user_id: Optional[str] = None
+    _analytics_opt_out: bool = False
+
+    def __new__(cls) -> "GlobalParameters":
+        """Singleton class definition.
```

Comment:
Nit: a more descriptive docstring, e.g., "Creates a singleton instance of the GlobalParameters class" or similar